hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -548,25 +548,34 @@ class SheetLine(models.TransientModel):
         default=0.0,
     )
 
-    @api.onchange('unit_amount')
-    def onchange_unit_amount(self):
-        """This method is called when filling a cell of the matrix.
-        It checks if there exists timesheets associated  to that cell.
-        If yes, it does several comparisons to see if the unit_amount of
-        the timesheets should be updated accordingly."""
-        self.ensure_one()
+    def _adjust_amount(self, value):
+        """For extensions"""
+        return value
 
+    def _get_sheet(self):
         sheet = self.sheet_id
         if not sheet:
             model = self.env.context.get('params', {}).get('model', '')
             obj_id = self.env.context.get('params', {}).get('id')
             if model == 'hr_timesheet.sheet' and isinstance(obj_id, int):
                 sheet = self.env['hr_timesheet.sheet'].browse(obj_id)
+        return sheet
+
+    @api.onchange('unit_amount')
+    def onchange_unit_amount(self):
+        """This method is called when filling a cell of the matrix.
+        It checks if there exists timesheets associated  to that cell.
+        If yes, it does several comparisons to see if the unit_amount of
+        the timesheets should be updated accordingly."""
+        self.ensure_one()
+        sheet = self._get_sheet()
         if not sheet:
             return {'warning': {
                 'title': _("Warning"),
                 'message': _("Save the Timesheet Sheet first.")
             }}
+        elif not self.sheet_id:
+            self.sheet_id = sheet
```

Comment:
I thought this is was not necessary, did you find a case where this is needed?

Reply:
Didn't checked. I just put it to be sure because I needed `self.sheet_id` in `_negative_unit_amount` method.

Reply:
removed those 2 lines...